### PR TITLE
Refactor tests, implement parallelized workers on CI

### PR DESCRIPTION
**Pull request by rtsao**
- Migrate tests to Jest
- Implements parallel containerized builds w/ Buildkite. Reduces CI test time from ~9 minutes to <3 minutes.

### CODE REVIEW

File: build/dev-runtime.js [hunk 1]
```diff
@@ -86,6 +86,8 @@ module.exports.DevelopmentRuntime = function(
   this.run = async function reloadProc() {
     const childPort = await getPort();
     const command = `
+      process.on('SIGTERM', () => process.exit());
```
Comment (KevinGrandon) [inline build/dev-runtime.js]: Why is this needed now, but not before? Just looking for some additional context.

File: test/e2e/assets-client-fetch/fixture/src/home.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// @flow
+// @noflow
```
Comment (KevinGrandon) [inline test/e2e/assets-client-fetch/fixture/src/home.js]: Why noflow? Can we $FlowFixMe instead? (Same for other noflow changes in next few files)

**Comment by rtsao** [comment]
Perhaps that's because Jest isn't showing the passing assertions (it only shows passing test files and failing assertions).

File: build/dev-runtime.js (hunk 1 above)
Comment (rtsao) [reply build/dev-runtime.js]: Without this, a zombie process was left each time dev was run.
